api/v1alpha1/hostedcluster_types.go

```diff
@@ -35,6 +35,20 @@ type HostedClusterSpec struct {
 
 	// InfraID is used to identify the cluster in cloud platforms
 	InfraID string `json:"infraID,omitempty"`
+
+	// DNS configuration for the cluster
```

Comment:
nit: isnt this for the ingress controller, not the cluster itself?

Reply:
@derekwaynecarr It is for the ingress controller. Which makes me think, should we allow you to specify an arbitrary list of ingress controllers like [Hive does](https://github.com/openshift/hive/blob/cacb68599e974a4e72d1bc8f1323ae16d370c102/apis/hive/v1/clusterdeployment_types.go#L100)?

Reply:
1. If this is for ingress, and if there are multiple ingresses configured (as @csrwng asks), does that mean each ingress's DNS zone is defined independently?
2. What fields are mutable? What is the behavior if they are mutated?

Reply:
Yes, this just configures the default ingress controller

Reply:
can we reflect this expectation either in a comment or the API field name?